src/Okta.Idx.Sdk/IdxClient.cs

```diff
@@ -237,6 +240,106 @@ internal async Task<IIdxResponse> IntrospectAsync(IIdxContext idxContext, Cancel
+        {
+            IIdxContext idxContext = await this.InteractAsync();
+            return new WidgetSignInResponse
+            {
+                IdxContext = idxContext,
+                SignInWidgetConfiguration = new SignInWidgetConfiguration(this.Configuration, idxContext),
+            };
+        }
+
+        /// <inheritdoc/>
+        public async Task<OktaTokens> RedeemInteractionCodeAsync(IIdxContext idxContext, string interactionCode, Action<Exception> exceptionHandler = null, CancellationToken cancellationToken = default)
+        {
+            exceptionHandler = exceptionHandler ?? LogError;
+            try
+            {
+                Uri issuerUri = new Uri(Configuration.Issuer);
+                Uri tokenUri = new Uri(GetNormalizedUriString(issuerUri.ToString(), "v1/token"));
+                HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Post, tokenUri);
+
+                StringBuilder requestContent = new StringBuilder();
+                this.AddParameter(requestContent, "grant_type", "interaction_code", false);
+                this.AddParameter(requestContent, "client_id", Configuration.ClientId, true);
+                if (!string.IsNullOrEmpty(Configuration.ClientSecret))
+                {
+                    this.AddParameter(requestContent, "client_secret", Configuration.ClientSecret, true);
+                }
+
+                this.AddParameter(requestContent, "interaction_code", interactionCode, true);
+                this.AddParameter(requestContent, "code_verifier", idxContext.CodeVerifier, true);
+
+                requestMessage.Content = new StringContent(requestContent.ToString(), Encoding.UTF8, "application/x-www-form-urlencoded");
+                requestMessage.Headers.Add("Accept", "application/json");
+                HttpResponseMessage responseMessage = await this.httpClient.SendAsync(requestMessage);
+                string tokenResponseJson = await responseMessage.Content.ReadAsStringAsync();
+
+                if (!responseMessage.IsSuccessStatusCode)
+                {
+                    exceptionHandler(new RedeemInteractionCodeException(tokenResponseJson));
+                }
+
+                return JsonConvert.DeserializeObject<OktaTokens>(tokenResponseJson);
+            }
+            catch (Exception exception)
+            {
+                exceptionHandler(new RedeemInteractionCodeException(exception));
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Logs the specified exception.
+        /// </summary>
+        /// <param name="ex">The exception.</param>
+        protected virtual void LogError(Exception ex)
+        {
+            _logger.LogError(ex, ex.Message);
+        }
+
+        private void AddParameter(StringBuilder stringBuilder, string key, string value, bool ampersandPrefix = false)
```

Comment:
Can we move these 3 utility methods to a helper so we can reuse them in other places? I can benefit from them in other classes :).

Reply:
It looks like there is currently a UrlHelper class in use here but it is defined in Okta.Sdk.Abstractions.  Let's talk briefly about our strategy here.